cdc/processor/processor.go

```diff
@@ -0,0 +1,737 @@
+			zap.Error(err))
+		return state, cerror.ErrReactorFinished
+	}
+	p.firstTick = false
+	return state, nil
+}
+
+func (p *processor) tick(ctx context.Context, state *changefeedState) (nextState orchestrator.ReactorState, err error) {
+	p.changefeed = state
+	if err := p.handleErrorCh(ctx); err != nil {
+		return nil, errors.Trace(err)
+	}
+	if p.changefeed.TaskStatus.AdminJobType.IsStopState() {
+		return nil, cerror.ErrAdminStopProcessor.GenWithStackByArgs()
+	}
+	if err := p.lazyInit(ctx); err != nil {
+		return nil, errors.Trace(err)
+	}
+	if skip := p.initPosition(); skip {
+		return p.changefeed, nil
+	}
+	if err := p.handleTableOperation(ctx); err != nil {
+		return nil, errors.Trace(err)
+	}
+	if err := p.initTables(ctx); err != nil {
+		return nil, errors.Trace(err)
+	}
+	if err := p.handlePosition(); err != nil {
+		return nil, errors.Trace(err)
+	}
+	if err := p.pushResolvedTs2Table(); err != nil {
+		return nil, errors.Trace(err)
+	}
+	if err := p.handleWorkload(); err != nil {
+		return nil, errors.Trace(err)
+	}
+	if err := p.doGCSchemaStorage(); err != nil {
+		return nil, errors.Trace(err)
+	}
+	return p.changefeed, nil
+}
+
+func (p *processor) initPosition() bool {
+	if p.changefeed.TaskPosition != nil {
+		return false
+	}
+	if !p.firstTick {
+		log.Warn("position is nil, maybe position info is removed unexpected", zap.Any("state", p.changefeed))
+	}
+	checkpointTs := p.changefeed.Info.GetCheckpointTs(p.changefeed.Status)
+	p.changefeed.PatchTaskPosition(func(position *model.TaskPosition) (*model.TaskPosition, error) {
+		if position == nil {
+			return &model.TaskPosition{
+				CheckPointTs: checkpointTs,
+				ResolvedTs:   checkpointTs,
+			}, nil
+		}
+		return position, nil
+	})
+	return true
+}
+
+func (p *processor) lazyInitImpl(ctx context.Context) error {
+	if !p.firstTick {
+		return nil
+	}
+	ctx, cancel := context.WithCancel(ctx)
+	p.cancel = cancel
+	ctx = util.PutChangefeedIDInCtx(ctx, p.changefeed.ID)
+
+	errCh := make(chan error, 16)
+	go func() {
+		// there are some other objects need errCh, such as sink and sink manager
+		// but we can't ensure that all the producer of errCh are non-blocking
+		// It's very tricky that create a goroutine to receive the local errCh
+		// TODO(leoppro): we should using `pkg/context.Context` instead of standard context and handle error by `pkg/context.Context.Throw`
+		for {
+			select {
+			case <-ctx.Done():
+				close(errCh)
+				return
+			case err := <-errCh:
+				if err == nil {
+					return
+				}
+				p.sendError(err)
+			}
+		}
+	}()
+
+	var err error
+	p.filter, err = filter.NewFilter(p.changefeed.Info.Config)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	p.schemaStorage, err = p.createAndDriveSchemaStorage(ctx)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	p.mounter = entry.NewMounter(p.schemaStorage, p.changefeed.Info.Config.Mounter.WorkerNum, p.changefeed.Info.Config.EnableOldValue)
+	go func() {
+		p.sendError(p.mounter.Run(ctx))
```

Comment:
Is there any plan to manage the goroutine, such as wait routine exit when the life cycle of processor is end.

Reply:
when `ctx` is done, this goroutine will be exit, can when processor is closed(call processor.Close()),the ctx will be done.